ompi/mca/mtl/base/mtl_base_datatype.h

```diff
@@ -125,138 +60,96 @@ ompi_mtl_datatype_pack(struct opal_convertor_t *convertor,
     }
     iov.iov_len = *buffer_len;
     iov.iov_base = NULL;
+    /* opal_convertor_need_buffers always returns true
+     * if CONVERTOR_ACCELERATOR is set, so unset temporarily
+     */
+    convertor->flags &= ~CONVERTOR_ACCELERATOR;
```

Comment:
This feels like the wrong approach to me.  We shouldn't go mucking with convertor flags because of an internal behavior of the converter interface; we should fix the convertor interface.

Reply:
I'll rethink this

Reply:
Fixed this